src/app/features/posts/containers/view-post/view-post.component.ts

```diff
@@ -25,12 +28,29 @@ export class ViewPostComponent {
 
   fetchComments = new EventEmitter<number>();
   addCommentEmitter = new EventEmitter<any>();
+  addReplyEmitter = new EventEmitter<{
+    reply: any;
+    comment: any;
+    component: NewCommentComponent;
+  }>();
+
   showFullSizeImage = false;
 
   comments: any = this.fetchComments.asObservable().pipe(
     startWith(''),
     switchMap(() => this.route.params.pipe(pluck('postId'))),
-    switchMap(postId => this.commentService.getByPostId(+postId, 1, 100))
+    switchMap(postId =>
+      this.commentService.getByPostIdWithReplies(+postId, 1, 100).pipe(
+        map(comments => {
+          comments.forEach(element => {
+            element.showReplies = false;
+            element.replyCount = element.commentReplies.length;
+          });
+
+          return comments;
```

Comment:
```typescript
comments.forEach(element => {
            element.showReplies = false;
            element.replyCount = element.commentReplies.length;
          });
```
element can be renamed to comment

Also, if the value returned is not changed (the reference is the same), tap operator is preferred instead of map.

Reply:
Another thing can be done to reduce nesting. 

The tap operator can be chained with the first pipe, the result of the request is passed to the next operator.